docs(trace-view): explain the new UI (#9519)
getsentry/sentry-docs: docs/product/sentry-basics/concepts/tracing/trace-view.mdx

description: "Learn more about trace view, where you can drill down into the details of a single trace, allowing you to debug slow services, identify related errors, and find other bottlenecks."
---

> Sentry provides a waterfall trace view similar to what you see in other APM solutions. The goal of this view is to show you everything that may have occurred during that trace, be it any errors, performance issues or profiles that may have been collected. The trace view is designed to be your one stop shop when debugging performance or errors, and it is the one place where you can gather full context on what was happening during an error or a performance issue. This experience is critical to us, so feedback (especially candid) is always appreciated.

Author: Good idea!

    allowfullscreen
    allow="clipboard-write"
    style={{
      colorScheme: "light",
      height: "100%",
      left: "0px",
      position: "absolute",
      top: "0px",
      width: "100%",
    }}
    title="Trace view - Tabbing"
  ></iframe>
</div>

Sometimes, duration gaps between entries can be very small, yet still very critical to your debugging effort, this is why the new view enables you to zoom into certain areas of the trace, either by double clicking on row elements, which zooms into the row you are scrolling to, or by using your mouse wheel.

<div
  style={{
    height: "0px",
    paddingBottom: "calc(66.66% + 41px)",
    position: "relative",
    width: "100%",
  }}
>
  <iframe
    src="https://demo.arcade.software/U8JbMjFJR1KRiP4G4j4E?embed&show_copy_link=true"
    title="Trace Details — Performance — sentry — Sentry"
    frameborder="0"
    loading="lazy"
    webkitallowfullscreen
    mozallowfullscreen
    allowfullscreen
    allow="clipboard-write"
    style={{
      colorScheme: "light",
      height: "100%",
      left: "0px",
      position: "absolute",
      top: "0px",
      width: "100%",
    }}
    title="Trace view - Zooming"
  ></iframe>
</div>

> The trace view has other features like search, which makes it easy for you to visualize matching spans and iterate over them by pressing the up or down arrow keys as well as tailoring the layout to your preferences. That said, this is not then end, and we are going to continue adding features that enable you to perform efficient investigation into whatever issues you are facing. If you have feedback or requests, please dont hesitate to reach out as we would love to hear how we can make your life easiers.

Reviewer: ```suggestion
The trace view has other features like search, which makes it easy to visualize matching spans that you can then iterate over using the up or down arrows. You can also tailor the layout to your preference. 

We're going to continue adding features that enable you to perform efficient investigation into whatever issues you might be facing. If you have feedback or requests, please share it here (ADD LINK). We'd love to hear how we can make your life easier and your users happier.
```
I think this is the perfect spot to ask for feedback. Would be good to add a link.

description: "Learn more about trace view, where you can drill down into the details of a single trace, allowing you to debug slow services, identify related errors, and find other bottlenecks."
---

The trace view is designed to be your one-stop-shop when debugging performance or errors. It gives you full context on what was happening when an error or performance issue occurred, all in one place. The waterfall trace view allows you to see everything that may have occurred during a trace, including errors, performance issues, and any profiles that may have been collected.

In addition, the trace view allows you to drill down into the details of a single trace so you can visualize the high level transactions and spans that took place within that trace. This makes debugging slow services, identifying related errors, and rooting out bottlenecks easer and faster without having to navigate around multiple pages of the Sentry product.

> ![Example of Trace View](./img/trace-view.png)

Reviewer 2: Should we update or remove this screenshot? I see later we have a walkthrough that's based on the new view

Author: Yup, definitely!

      width: "100%",
    }}
    title="Trace view - Tabbing"
  ></iframe>
</div>

Sometimes, duration gaps between entries can be small, but critical to your debugging effort. That's why the trace view enables you to zoom into certain areas of the trace, either by double clicking on row elements, (which zooms into the row), or by using your mouse wheel.

<div
  style={{
    height: "0px",
    paddingBottom: "calc(66.66% + 41px)",
    position: "relative",
    width: "100%",
  }}
>
  <iframe
    src="https://demo.arcade.software/U8JbMjFJR1KRiP4G4j4E?embed&show_copy_link=true"
    title="Trace Details — Performance — sentry — Sentry"
    frameborder="0"
    loading="lazy"
    webkitallowfullscreen
    mozallowfullscreen
    allowfullscreen
    allow="clipboard-write"
    style={{
      colorScheme: "light",
      height: "100%",
      left: "0px",
      position: "absolute",
      top: "0px",
      width: "100%",
    }}
    title="Trace view - Zooming"
  ></iframe>
</div>

The trace view has other features like search, which makes it easy to visualize matching spans that you can then iterate over using the up or down arrows. You can also tailor the layout to your preference. 

> We're going to continue adding features that enable you to perform efficient investigation into whatever issues you might be facing. If you have feedback or requests, please share it here (ADD LINK). We'd love to hear how we can make your life easier and your users happier.

Reviewer 2: Just commenting to make sure we remember to `(ADD LINK)`

description: "Learn more about trace view, where you can drill down into the details of a single trace, allowing you to debug slow services, identify related errors, and find other bottlenecks."
---

The trace view is designed to be your one-stop-shop when debugging performance or errors. It gives you full context on what was happening when an error or performance issue occurred, all in one place. The waterfall trace view allows you to see everything that may have occurred during a trace, including errors, performance issues, and any profiles that may have been collected.

> In addition, the trace view allows you to drill down into the details of a single trace so you can visualize the high level transactions and spans that took place within that trace. This makes debugging slow services, identifying related errors, and rooting out bottlenecks easer and faster without having to navigate around multiple pages of the Sentry product.

Reviewer 3: ```suggestion
In addition, the trace view allows you to drill down into the details of a single trace so you can visualize the high-level transactions and spans that took place within that trace. This makes debugging slow services, identifying related errors, and rooting out bottlenecks easier and faster, since you no longer have to navigate around multiple pages of the Sentry product.
```

## Product Walkthrough: Trace View Page

> Sentry's default visualization of a trace is a waterfall-like structure, where the entries in the list are organized by chronological order as well as ancestry. This allows you to follow the order of operations as well as track sub-operations.

Reviewer 3: ```suggestion
Sentry's default visualization of a trace is a waterfall-like structure, where the entries (transactions and spans) in the list are organized chronologically and through ancestry (child spans will fall under their parents). This allows you to follow the order of operations and drill into sub-operations.
```

On the left side is a list of operations, and on the right is their duration and any events, such as errors, which may have occurred in that timeframe.

> By default, the waterfall view shows a list of transactions. If you need more granular data and want to see spans as well, click the + button on the left and the waterfall will update and show both. The waterfall view helps surface any performance issues or profiles collected during a given timeframe.

Reviewer 3: ```suggestion
By default, the waterfall view shows a list of transactions. If you need more granular data and want to see spans as well, click "+" to the left of a transaction to expand the waterfall. The waterfall view helps surface any performance issues or profiles collected during a given timeframe.
```
I'm assuming from the arcade that the + only expands to show spans for that transaction, not all of them in the trace right?

Whatever you highlight will also be saved if you navigate away from the page and will still be there when you use your browser's back and forward buttons.

If you're doing a comparison and want an easy way to go back and forth between highlighted areas of interest in your trace, you can pin the tabs. When a tab is pinned, the view will be persistent throughout your session.

<br />
<div
  style={{
    height: "0px",
    paddingBottom: "calc(66.66% + 41px)",
    position: "relative",
    width: "100%",
  }}
>
  <iframe
    src="https://demo.arcade.software/yP3PFytK0ckfetjJTe52?embed&show_copy_link=true"
    title="Sentry"
    frameborder="0"
    loading="lazy"
    webkitallowfullscreen
    mozallowfullscreen
    allowfullscreen
    allow="clipboard-write"
    style={{
      colorScheme: "light",
      height: "100%",
      left: "0px",
      position: "absolute",
      top: "0px",
      width: "100%",
    }}
    title="Trace view - Tabbing"
  ></iframe>
</div>
<br />
Sometimes, duration gaps between entries can be small, but critical to your debugging
effort. That's why the trace view enables you to zoom into certain areas of the trace,

> either by double clicking on row elements, (which zooms into the row), or by using
> your mouse wheel.

Reviewer 3: ```suggestion
either by double clicking on row elements (which zooms into the row) or by using
your scroll wheel.
```
I think scroll wheel is a more agnostic term in case users are just on a laptop without a mouse

<br />
<div
  style={{
    height: "0px",
    paddingBottom: "calc(66.66% + 41px)",
    position: "relative",
    width: "100%",
  }}
>
  <iframe
    src="https://demo.arcade.software/U8JbMjFJR1KRiP4G4j4E?embed&show_copy_link=true"
    title="Trace Details — Performance — sentry — Sentry"
    frameborder="0"
    loading="lazy"
    webkitallowfullscreen
    mozallowfullscreen
    allowfullscreen
    allow="clipboard-write"
    style={{
      colorScheme: "light",
      height: "100%",
      left: "0px",
      position: "absolute",
      top: "0px",
      width: "100%",
    }}
    title="Trace view - Zooming"
  ></iframe>
</div>
<br />
The trace view has other features like search, which makes it easy to visualize matching
spans that you can then iterate over using the up or down arrows. You can also tailor
the layout to your preference.

> We're going to continue adding features that enable you to perform efficient investigation into whatever issues you might be facing. If you have feedback or requests, please share it with us. We'd love to hear how we can make your life easier and your users happier.

Reviewer 3: Would be great to add a link to give feedback here if we're inviting users to do so

Author: Where should we link @vivianyentran ?
